addons/sale_loyalty/models/sale_order.py

```diff
@@ -7,15 +7,29 @@
 
 import random
 
+from pytz import timezone
+
 from odoo import api, fields, models, _
 from odoo.exceptions import UserError, ValidationError
 from odoo.fields import Command
 from odoo.tools.float_utils import float_is_zero, float_round
 from odoo.osv import expression
 
+
 def _generate_random_reward_code():
     return str(random.getrandbits(32))
 
+
+def _get_today_from_order(order):
```

Comment:
```suggestion
def _get_confirmed_tx_creation_date(order):
```
Maybe also could be useful to add docstring explaining why we are taking "past" date for loyalty domain.